pkg/kubectl/cmd/util/factory.go

```diff
@@ -106,8 +108,23 @@ func NewFactory(optionalClientConfig clientcmd.ClientConfig) *Factory {
 		clientConfig = DefaultClientConfig(flags)
 	}
 
-	clients := NewClientCache(clientConfig)
+	clients := newClientCache(clientConfig)
```

Comment:
I think it should be public.  Other people may want a client cache.

On Fri, Aug 7, 2015 at 5:16 PM, Muhammed Uluyol <notifications@github.com>
wrote:

> In pkg/kubectl/cmd/util/factory.go
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/12405#discussion_r36561991>
> :
>
> > @@ -106,8 +108,23 @@ func NewFactory(optionalClientConfig clientcmd.ClientConfig) *Factory {
> >  		clientConfig = DefaultClientConfig(flags)
> >  	}
> >
> > -	clients := NewClientCache(clientConfig)
> > +	clients := newClientCache(clientConfig)
>
> Then we really should either make clientCache public or make an interface
> and return that. I don't think NewClientCache should be public while
> returning a private type.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/12405/files#r36561991>
> .
>



-- 
Clayton Coleman | Lead Engineer, OpenShift